audit/helper.go

```diff
@@ -0,0 +1,51 @@
+package audit
+
+import (
+	"context"
+	"strings"
+
+	"github.com/ONSdigital/go-ns/common"
+)
+
+// pathIDs relate the possible string matches in path with the respective
+// parameter name as key values pairs
+var pathIDs = map[string]string{
+	"jobs":                  "job_id",
+	"datasets":              "dataset_id",
+	"editions":              "edition",
+	"versions":              "version",
+	"dimensions":            "dimension",
+	"options":               "option",
+	"instances":             "instance_id",
+	"inserted_observations": "inserted_observations",
+	"node_id":               "node_id",
+	"filters":               "filter_blueprint_id",
+	"filter-outputs":        "filter_output_id",
+	"hierarchies":           "instance_id",
+	// TODO add params for codelist API endpoints
+}
+
+// GetParameters populates audit parameters with path variable values
+func GetParameters(ctx context.Context, path string) common.Params {
+	auditParams := common.Params{}
+
+	callerIdentity := common.Caller(ctx)
+	if callerIdentity != "" {
+		auditParams["caller_identity"] = callerIdentity
+	}
+
+	splitPath := strings.Split(path, "/")
+
+	for i, pathValue := range splitPath {
+		if i+1 >= len(splitPath) {
+			break
+		}
+
+		parameter, ok := pathIDs[pathValue]
+		if ok {
+			auditParams[parameter] = splitPath[i+1]
+		}
```

Comment:
how about adding else { auditParams[pathValue] = splitPath[i+1] } (i.e. default to pathValue if that key doesn't exist in the map - and then you can remove the 2 key-value pairs above where they match (and future expansion defaults to using the existing pathValue and doesn't go undetected)?

Reply:
We can't assume the structure of the url will always be the same;

Most of our urls are like so:

`/<lists>/<id>/<lists>/<id>.....`

But there are some different structures like the following:

`/search/datasets/<id>` and `/hierarchies/<instance>/<dimension>`

It is difficult to capture all these without creating a list; a map is used to know when an id is a `dataset_id`, `instance_id` etc.